play.pokemonshowdown.com/src/battle-team-editor.tsx

```diff
@@ -675,6 +675,33 @@ class TeamEditorState extends PSModel {
 		};
 		return Object.keys(all);
 	}
+	/** returns null if no boxes exist, empty array if no sets for this species */
+	getUserSets(set: Dex.PokemonSet): { [setName: string]: Dex.PokemonSet } | null {
+		const userSets: { [species: string]: { [setName: string]: Dex.PokemonSet } } = {};
+		let hasBoxes = false;
+
+		for (const team of window.PS?.teams.list || []) {
+			if (team.format !== this.format || !team.isBox) continue;
+			hasBoxes = true;
+
+			const setList = Teams.unpack(team.packedTeam);
```

Comment:
Okay so it looks like we're unpacking every box that matches this format. Seems valid but we should probably cache the result?